layer/contracts/definitions.py

```diff
@@ -80,32 +68,20 @@ def _get_source(self) -> str:
         except Exception:
             return "source code not available"
 
-    @property
-    def project_full_name(self) -> ProjectFullName:
-        return ProjectFullName(
-            account_name=self.account_name,
-            project_name=self.project_name,
-        )
-
     @property
     def asset_path(self) -> AssetPath:
+        project_full_name = context.get_project_full_name()
```

Comment:
why do they need to be taken out of constructor / properties?

invoking `context` from deep within like will create headaches down the road imo
-> said another way: is there really a situation when the `FunctionDefinition` is already done (instantiated), and we change the account/project before invoking it?

Why dont we use `context.get_project_full_name()` at creation time?

Reply:
good point, i think there is no use case where context changes once `FunctionDefinition` is initialized, so we can simplify things that way.

there is some other refactoring that I'd like to do for `FunctionDefinition` as well, once we switch over to using functions for remote execution only (as opposed to using it for experiment tracking and artifact storage along with remote execution). mind if we leave those changes till then?